### ISSUE TITLE
Talon crashed with ZeroDivisionError

### ISSUE THREAD

**Issue by wolfmanstout** [talonvoice/talon#393]
I saw this error upon returning to my computer:
![div_zero](https://user-images.githubusercontent.com/1121562/137248774-d338430a-11f5-409b-a64b-d79c3ee63640.png)

From the logs:
```
2021-10-13 17:04:25    IO Exception ignored from cffi callback <function screen_event at 0x0000000003E6C8B0>:
2021-10-13 17:04:25    IO Traceback (most recent call last):
2021-10-13 17:04:25    IO   File "talon\screen.py", line 207, in screen_event
2021-10-13 17:04:25    IO   File "talon\screen.py", line 161, in from_c
2021-10-13 17:04:25    IO   File "talon\screen.py", line 165, in _recompute_dpi
2021-10-13 17:04:25    IO ZeroDivisionError: float division by zero
2021-10-13 17:04:25    IO Exception ignored from cffi callback <function screen_event at 0x0000000003E6C8B0>:
2021-10-13 17:04:25    IO Traceback (most recent call last):
2021-10-13 17:04:25    IO   File "talon\screen.py", line 207, in screen_event
2021-10-13 17:04:25    IO   File "talon\screen.py", line 161, in from_c
2021-10-13 17:04:25    IO   File "talon\screen.py", line 165, in _recompute_dpi
2021-10-13 17:04:25    IO ZeroDivisionError: float division by zero
2021-10-13 17:04:26    IO Exception ignored from cffi callback <function screen_event at 0x0000000003E6C8B0>:
2021-10-13 17:04:26    IO Traceback (most recent call last):
2021-10-13 17:04:26    IO   File "talon\screen.py", line 207, in screen_event
2021-10-13 17:04:26    IO   File "talon\screen.py", line 161, in from_c
2021-10-13 17:04:26    IO   File "talon\screen.py", line 165, in _recompute_dpi
2021-10-13 17:04:26    IO ZeroDivisionError: float division by zero
2021-10-13 17:04:29 WARNING [watchdog] "talon.windows.ui.proxy_screen_change" @2.0s
   18:                threading.py:930* # cron thread
   17:                threading.py:973* 
   16:                threading.py:910* 
   15:               talon\cron.py:155| 
   14:               talon\cron.py:106| 
   13: talon\scripting\dispatch.py:99 | 
   12: talon\scripting\dispatch.py:105| 
   11: talon\scripting\dispatch.py:144| 
   10: talon\scripting\dispatch.py:133| 
    9:     talon\scripting\rctx.py:233| # 'screen_update' main:proxy_screen_change()
    8:         talon\windows\ui.py:385| 
    7: talon\scripting\dispatch.py:105| 
    6: talon\scripting\dispatch.py:144| 
    5: talon\scripting\dispatch.py:135| 
    4:     talon\scripting\rctx.py:233| # 'screen_change' talon_plugins.eye_mouse:on_screen_change()
    3:  talon_plugins\eye_mouse.py:573| tracker.display_setup(config.size_mm.x..
    2:        talon\track\tobii.py:407| 
    1:        talon\track\tobii.py:329| 
2021-10-13 17:04:29 WARNING [watchdog] "talon_plugins.eye_mouse.on_screen_change" @2.0s
   18:                threading.py:930* # cron thread
   17:                threading.py:973* 
   16:                threading.py:910* 
   15:               talon\cron.py:155| 
   14:               talon\cron.py:106| 
   13: talon\scripting\dispatch.py:99 | 
   12: talon\scripting\dispatch.py:105| 
   11: talon\scripting\dispatch.py:144| 
   10: talon\scripting\dispatch.py:133| 
    9:     talon\scripting\rctx.py:233| # 'screen_update' main:proxy_screen_change()
    8:         talon\windows\ui.py:385| 
    7: talon\scripting\dispatch.py:105| 
    6: talon\scripting\dispatch.py:144| 
    5: talon\scripting\dispatch.py:135| 
    4:     talon\scripting\rctx.py:233| # 'screen_change' talon_plugins.eye_mouse:on_screen_change()
    3:  talon_plugins\eye_mouse.py:573| tracker.display_setup(config.size_mm.x..
    2:        talon\track\tobii.py:407| 
    1:        talon\track\tobii.py:329| 
2021-10-13 17:04:31 WARNING [watchdog] "talon.windows.ui.proxy_screen_change" @4.0s (stalled)
2021-10-13 17:04:31 WARNING [watchdog] "talon_plugins.eye_mouse.on_screen_change" @4.0s (stalled)
2021-10-13 17:04:33 WARNING [watchdog] "talon.windows.ui.proxy_screen_change" @6.0s (stalled)
2021-10-13 17:04:33 WARNING [watchdog] "talon_plugins.eye_mouse.on_screen_change" @6.0s (stalled)
2021-10-13 17:04:35 WARNING [watchdog] "talon.windows.ui.proxy_screen_change" @8.0s (stalled)
2021-10-13 17:04:35 WARNING [watchdog] "talon_plugins.eye_mouse.on_screen_change" @8.0s (stalled)
2021-10-13 17:04:37 WARNING [watchdog] "talon.windows.ui.proxy_screen_change" @10.0s (stalled)
2021-10-13 17:04:37 ERROR cb error topic="screen_change" cb=<function on_screen_change at 0x000000000772FC10>
   19:                threading.py:930* # cron thread
   18:                threading.py:973* 
   17:                threading.py:910* 
   16:               talon\cron.py:155| 
   15:               talon\cron.py:106| 
   14: talon\scripting\dispatch.py:99 | 
   13: talon\scripting\dispatch.py:105| 
   12: talon\scripting\dispatch.py:144| 
   11: talon\scripting\dispatch.py:133| 
   10:     talon\scripting\rctx.py:233| # 'screen_update' main:proxy_screen_change()
    9:         talon\windows\ui.py:385| 
    8: talon\scripting\dispatch.py:105| 
    7: talon\scripting\dispatch.py:144| 
    6: talon\scripting\dispatch.py:135| 
    5: -------------------------------# cron thread
    4:     talon\scripting\rctx.py:233| # 'screen_change' talon_plugins.eye_mouse:on_screen_change()
    3:  talon_plugins\eye_mouse.py:573| tracker.display_setup(config.size_mm.x..
    2:        talon\track\tobii.py:407| 
    1:        talon\track\tobii.py:359| 
talon.track.tobii.EyeClosedErr: Eye Tracker timed out or disconnected
2021-10-13 17:05:44    IO [-] USB: Tobii 5 detached
2021-10-13 17:05:44 ERROR cb error topic="detach" cb=<bound method TrackingSystem.on_detach of <talon.track.system.TrackingSystem object at 0x0000000005B2E9F0>>
   13:                threading.py:930* # cron thread
   12:                threading.py:973* 
   11:                threading.py:910* 
   10:               talon\cron.py:155| 
    9:               talon\cron.py:106| 
    8:                talon\usb.py:65 | 
    7: talon\scripting\dispatch.py:105| 
    6: talon\scripting\dispatch.py:144| 
    5: talon\scripting\dispatch.py:135| 
    4: -------------------------------# cron thread
    3:     talon\scripting\rctx.py:233| # 'detach' main:on_detach()
    2:       talon\track\system.py:42 | 
    1:       talon\track\system.py:66 | 
KeyError: <TobiiEC 578024688 "Tobii 5">
```

**Comment by wolfmanstout** [comment]
Version v0.2.0-114-g71c3 (114).

**Comment by lunixbochs** [comment]
when you say crash, talon was still running after this right? it was just a msgbox?

**Comment by wolfmanstout** [comment]
I do distinctly recall this was a crash. I remember the Talon icon in the notification stuck around until I hovered the mouse over it and it disappeared. I can't repro it to prove that though so it's possible I'm remembering incorrectly.

**Comment by wolfmanstout** [comment]
I see a couple logs after that I left out, and then the next one is from a fresh Talon invocation:
```
2021-10-13 17:05:46    IO [+] USB: Tobii 5 attached
2021-10-13 17:05:46  INFO Eye Tracker Firmware: 10008
```

**Comment by lunixbochs** [comment]
Ok sure, that ZeroDivisionError wasn't the cause of the crash though. Can you pull up event viewer and see the crashing module / offset? (Even better if you have crash dumps enabled on your system and can get a full stack trace)

**Comment by wolfmanstout** [comment]
I assume this would be under Windows Logs > Application? I don't see any crash logs there. I do see that I had updated my graphics card drivers at around this time, however. Sorry for not mentioning that earlier;  I kicked this off in the background and then came back to the computer later to see this error.

**Comment by lunixbochs** [comment]
FWIW
```
2021-10-13 17:05:46    IO [+] USB: Tobii 5 attached
2021-10-13 17:05:46  INFO Eye Tracker Firmware: 10008
```
doesn't indicate a fresh Talon invocation, the first lines in the log on Windows are about hotplug and Talon version

**Comment by wolfmanstout** [comment]
I meant that those were the couple logs I left out from the invocation that crashed. Here's the  sequence I see in the logs:
```
2021-10-13 17:05:46    IO [+] USB: Tobii 5 attached
2021-10-13 17:05:46  INFO Eye Tracker Firmware: 10008
2021-10-13 20:45:52  INFO Activating Microphone: "Headset Microphone (Sennheiser SD for Lync)"
2021-10-13 20:45:53  INFO Falling back to polling for USB hotplug: failed to register hotplug callback
```

**Comment by lunixbochs** [comment]
I'm closing this because I believe I just fixed the ZeroDivisionError in 0.2.0-116 (and it was a basic python exception, so it's very unlikely it would crash the parent app), but please keep a look out for the crash and open a new issue if you hit it again.

It's weird there's no trace of a crash, maybe that happens if you have a JIT debugger? It might be helpful for you to install the Windbg Preview preemptively and try to get a stack trace if a crash does happen.

Otherwise I might need to set up Crashpad or something so Talon can log crashes on Windows itself.
